spectral_cube/analysis_utilities.py

```diff
@@ -341,39 +347,88 @@ def stack_cube(cube, linelist, vmin, vmax, average=np.nanmean, convolve_beam=Non
         If the cube is a VaryingResolutionSpectralCube, a convolution beam is
         required to put the cube onto a common grid prior to spectral
         interpolation.
+    return_hdu : bool
+        Return an HDU instead of a spectral-cube
+    return_cutouts : bool
+        Also return the individual cube cutouts?
+
+    Returns
+    =======
+    cube : SpectralCube
+        The SpectralCube object containing the reprojected cube.  Its header
+        will be based on the first cube but will have no reference frequency.
+        Its spectral axis will be in velocity units.
+    cutout_cubes : list
+        A list of cube cutouts projected into the same space (optional; see
+        ``return_cutouts``)
     """
 
-    line_cube = cube.with_spectral_unit(u.km/u.s,
-                                        velocity_convention='radio',
-                                        rest_value=linelist[0])
-    if isinstance(line_cube, VaryingResolutionSpectralCube):
-        if convolve_beam is None:
-            raise ValueError("When stacking VaryingResolutionSpectralCubes, "
-                             "you must specify a target beam size with the "
-                             "keyword `convolve_beam`")
-        reference_cube = line_cube.spectral_slab(vmin, vmax).convolve_to(convolve_beam)
+    if isinstance(cube, list):
+        cubes = cube
+        cube = cubes[0]
+        for cb in cubes[1:]:
+            if cb.shape[1:] != cube.shape[1:]:
+                raise ValueError("If you pass multiple cubes, they must have the "
+                                 "same spatial shape.")
+        if convolve_beam is None and (any(hasattr(cb, 'beams') for cb in cubes) or
+                                      len(set([cb.beam for cb in cubes])) > 1):
```

Comment:
```suggestion
                                      not all(cb.beam == cubes[0].beam for cb in cubes):
```

Test failures from here: https://github.com/radio-astro-tools/spectral-cube/runs/4268504713?check_suite_focus=true#step:5:314.

Reply:
fixed; pushed - running tests locally too